lib/diagnostics_channel.js

```diff
@@ -136,10 +136,95 @@ function hasSubscribers(name) {
   return channel.hasSubscribers;
 }
 
+class TracingChannel {
+  constructor(name) {
+    this.name = name;
+    this.channels = {
+      start: channel(`${name}.start`),
+      end: channel(`${name}.end`),
+      asyncEnd: channel(`${name}.asyncEnd`),
+      error: channel(`${name}.error`)
```

Comment:
Wondering if we need to declare four channels by default.

Maybe we could create channels on the fly each time the `subscribe` method is invoked.

Reply:
We need them all for the publishing side and it's best to create the channels out-of-band from possibly hot code paths where the publishes happen.